In [1]:
import pandas as pd
import matplotlib.pyplot as plt

#Load the data

In [2]:
df=pd.read_csv('/content/Age_Height.csv')

In [3]:
df.head()

Age  Height
0   10     138
1   11     138
2   12     138
3   13     139
4   14     139

In [4]:
df.tail()

Age  Height
66   76     204
67   77     205
68   78     206
69   79     207
70   80     208

In [5]:
df.shape

(71, 2)

Splitting Dependent and independent data

In [6]:
X = df.iloc[:,0:1].values
y = df.iloc[:,1].values

Splitting into train and test dataset

In [7]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state=30)

#Creating Decision tree model and training

In [8]:
from sklearn.tree import DecisionTreeRegressor
model=DecisionTreeRegressor()
model.fit(X_train,y_train)


DecisionTreeRegressor()

In [9]:
y_pred=model.predict(X_test)

In [10]:
y_pred

array([144., 168., 159., 141., 146., 185., 206., 172., 163., 180., 144.,
       190., 147., 150., 195.])

#Evaluating the model

In [11]:
from sklearn.metrics import r2_score
print('R2 score for Polynomial Regression:',r2_score(y_test,y_pred))

R2 score for Polynomial Regression: 0.9924455205811138


#Predicting the result

In [12]:
df.sample(1)

Age  Height
45   55     164

In [13]:
model.predict([[64]])

array([172.])

In [14]:
df.sample(2)

Age  Height
68   78     206
25   35     147

In [15]:
model.predict([[53]])

array([163.])

The R2 score of 0.992 reflects an excellent fit of the model to the test data, indicating its ability to capture the relationship between age and height.
When predicting a height for an age of 64, the model estimated 172 cm.
When predicting a height for an age of 53, the prediction was 163 cm.
